xds/internal/resolver/watch_service.go

```diff
@@ -22,17 +22,28 @@ import (
 	"fmt"
 	"strings"
 	"sync"
+	"time"
 
 	"google.golang.org/grpc/internal/grpclog"
 	xdsclient "google.golang.org/grpc/xds/internal/client"
 )
 
-// serviceUpdate contains information received from the RDS responses which is
-// of interested to the xds resolver. The RDS request is built by first making a
-// LDS to get the RouteConfig name.
+// serviceUpdate contains information received from the LDS/RDS responses which
+// are of interest to the xds resolver. The RDS request is built by first
+// making a LDS to get the RouteConfig name.
 type serviceUpdate struct {
-	// Routes contain matchers+actions to route RPCs.
-	Routes []*xdsclient.Route
+	// routes contain matchers+actions to route RPCs.
+	routes []*xdsclient.Route
+	// ldsConfig contains configuration that applies to all routes.
+	ldsConfig ldsConfig
```

Comment:
Inline `maxStreamDuration` instead of making another struct?

Like we didn't put `routes` inside a `rdsConfig` struct.

Reply:
I think we should leave this since we are expecting more data in the LDS config (for fault injection), and we have to have logic to see if anything changed (line 109).